jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -2604,4 +2603,9 @@ else if (oldQuery == null)
     {
         throw new ServletException("HttpServletRequest.upgrade() not supported in Jetty");
     }
+
+    private String formatAddrOrHost(String name)
+    {
+        return _channel == null ? HostPort.normalizeHost(name) : _channel.formatAddrOrHost(name);
```

Comment:
`_channel` is a `final` field. Can it ever be `null`?
If it is only `null` in tests, you may want to add a comment about that.

Reply:
It should only be null in tests, but we have a specific test to check it works with null.